File: Java/State.java
public class Context {
	private State state;
	
	public Context() {
		this.state = new DefaultState();
	}
	
	public void changeState(State state) {
		this.state = state;
	}
	
	public void behavior() {
		this.state.behavior();
	}
}

public abstract class State {
	public abstract void behavior();
	
}

public class DefaultState implements State {
	public void behavior() {
		System.out.println("Hello World!")
	}
}

public class CustomState implements State {
	private Integer data = 0;
	
	public void behavior() {
		if (data++ < 9) {
			System.out.print("Na ");
		}
		else {
			System.out.println("Thunder!");
			data = 0;
		}
	}
}

File: Python/State.py
class Context():
	_state = None;
	
	def __init__(self):
		self.state = DefaultState();
	
	def changeState(state):
		_state = state;
	
	def behavior():
		_state.behavior();
	

class State():
	@abs.abstractmethod
	def behavior(self):
		pass;


class DefaultState(State):
	def behavior(self):
		print("Hello World!");


class CustomState(State):
	_data = 0;
	
	def behavior(self):
		if (data++ < 9):
			print("Na");
		else:
			print("Thunder!\n");
			data = 0;
